File: image/DrawLine/C++/OpenImageIODrawLine/DrawLine.cpp
// clang++ -std=c++14 DrawLine.cpp -lOpenImageIO -lOpenImageIO_Util -o DrawLine
#include <memory>
#include <string>
#include <OpenImageIO/imageio.h>
#include <OpenImageIO/imagebuf.h>
#include <OpenImageIO/imagebufalgo.h>



int main( int argc, char* argv[] )
{
  constexpr unsigned int           WIDTH     = 100;
  constexpr unsigned int           HEIGHT    = 100;
  using namespace OIIO;

  const int channels = 3; // RGB
  ImageSpec spec (WIDTH, HEIGHT, channels, TypeDesc::UINT8);
  auto buf=ImageBuf(spec);
  // note the algorithms use float values for parameters, here we create an array of 1,1,1 
  ImageBufAlgo::fill(buf,{1.0f,1.0f,1.0f});
//  This uses the same dimensions and co-ordinates as the image
//  first draw from top left to bottom right using red

  ImageBufAlgo::render_line(buf, 0, 0, WIDTH, HEIGHT, {1.0f,0.0f,0.0f});
  // now the other diagonal in green
  ImageBufAlgo::render_line(buf, WIDTH, 0, 0, HEIGHT, {0.0f,1.0f,0.0f});
  // horizontal  in blue
  ImageBufAlgo::render_line(buf, 0,HEIGHT/2,WIDTH,HEIGHT/2, {0.0f,0.0f,1.0f});
  // vertical in  black
  ImageBufAlgo::render_line(buf, WIDTH/2,0,WIDTH/2,HEIGHT, {0.0f,0.0f,0.0f});

  buf.write("DrawLine.png");
}

File: image/DrawLine/Python/OpenImageIODrawLine/DrawLine.py
#!/usr/bin/env python

from __future__ import print_function # python 3 style printing
import OpenImageIO as oiio

width=100
height=100
channels=3
# build our image specification
spec= oiio.ImageSpec(width,height, channels,oiio.UINT8)
# create a buffer for out image data
buf=oiio.ImageBuf(spec)

oiio.ImageBufAlgo.fill (buf, (255,255,255))
# This uses the same dimensions and co-ordinates as the image
# first draw from top left to bottom right using red

oiio.ImageBufAlgo.render_line(buf, 0, 0, width, height, (255,0,0))
# now the other diagonal in green
oiio.ImageBufAlgo.render_line(buf, width, 0, 0, height, (0,255,0))
# horizontal  in blue
oiio.ImageBufAlgo.render_line(buf, 0,image.height/2,image.width,image.height/2, (0,0,255))
# vertical in  black
oiio.ImageBufAlgo.render_line(buf, image.width/2,0,image.width/2,image.height, (255,0,0))





